src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DiagnosticEventLogger : ILogger
+    {
+        private const string ErrorCode = "errorCode";
+        private const string HelpLink = "helpLink";
+        private readonly IDiagnosticEventRepositoryFactory _diagnosticEventRepositoryFactory;
+        private readonly IEnvironment _environment;
+        private IDiagnosticEventRepository _diagnosticEventRepository;
+        private Lazy<bool> isDiagnosticEventLoggingEnabled = new Lazy<bool>(() => FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableDiagnosticEventLogging));
```

Comment:
Rather than doing this here, can we do it during service registration? So that this is off, the services themselves are never even registered?